### ISSUE TITLE
How to get this script to run from cron? - with solution

### ISSUE THREAD

**Issue by AntoniosHadji** [Anthony25/gnome-terminal-colors-solarized#68]
I was trying to get this to run via cron so the colors would change automatically.  dconf doesn't work unless the environment contains $DBUS_SESSION_BUS_ADDRESS.  cron is a very minimal environment and does not include this variable.  I found this [stack overflow answer ](http://stackoverflow.com/questions/10374520/gsettings-with-cron/19666729#19666729) which helped get it working on Ubuntu 16.04.  I added the following three lines to a streamlined script and it's currently working as expected.  

```
export DISPLAY=:0
PID=$(pgrep gnome-terminal)
export DBUS_SESSION_BUS_ADDRESS=$(grep -z DBUS_SESSION_BUS_ADDRESS /proc/$PID/environ|cut -d= -f2-)
```
Adding these three lines to the top of install.sh will let the set_light and set_dark scripts be called from cron.  I tested only on Ubuntu 16.04

**Comment by Anthony25** [comment]
Hello,
Thanks for this info! To be honest, I would prefer to avoid adding these lines into my script, as it could cause more problems than it would solve. For example, if your script is triggered by cron and no gnome-terminal is running, I do not think it will work, as `pgrep gnome-terminal` will return nothing.

But what I can do (and will do) is to add this info in the `README`. I will add some details, so users who want to use this script in cron will be aware about what they are doing, but it will avoid some black magic hidden into my scripts.

**Comment by AntoniosHadji** [comment]
Yes, this isn't really an issue.  I just spent a lot of time figuring out and thought it might be useful to others.  I'm still working on the issue you mention.  This morning the code did not execute because there was no gnome-session or gnome-terminal running when the cron job went off.  I'm looking at using systemd so the code only runs on login / logout.  I learned a lot figuring out what your script was doing.  Thank you for providing this.
